src/shogun/classifier/NearestCentroid.cpp

```diff
@@ -100,12 +92,10 @@ namespace shogun{
 			SGVector<float64_t>::scale_vector(scale,target,num_feats);
 		}
 
-		m_centroids->free_feature_matrix();
-		m_centroids->set_feature_matrix(centroids);
-
+		auto centroids_feats = some<CDenseFeatures<float64_t>>(centroids);
 
 		m_is_trained=true;
-		distance->init(m_centroids,distance->get_rhs());
+		distance->init(centroids_feats, distance->get_rhs());
```

Comment:
why the change here?

Reply:
since we would like to drop setters in features`set_num_vectors`, `set_num_features`, we can't an empty features and then set feature matrix later, and here there is no need to store it in a member field